src/main/scala/com/stratio/receiver/RabbitMQInputDStream.scala

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (C) 2015 Stratio (http://stratio.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.stratio.receiver
+
+import java.net.ConnectException
+import java.util
+
+import com.rabbitmq.client.{Channel, Connection, ConnectionFactory, QueueingConsumer}
+import org.apache.spark.Logging
+import org.apache.spark.storage.StorageLevel
+import org.apache.spark.streaming.StreamingContext
+import org.apache.spark.streaming.dstream.ReceiverInputDStream
+import org.apache.spark.streaming.receiver.Receiver
+
+
+private[receiver]
+class RabbitMQInputDStream(
+                         @transient ssc_ : StreamingContext,
+                         rabbitMQQueueName: Option[String],
+                         rabbitMQHost: String,
+                         rabbitMQPort: Int,
+                         exchangeName: Option[String],
+                         routingKeys: Seq[String],
+                         storageLevel: StorageLevel
+                         ) extends ReceiverInputDStream[String](ssc_) with Logging {
+
+  override def getReceiver(): Receiver[String] = {
+    val DefaultRabbitMQPort = 5672
+
+    new RabbitMQReceiver(rabbitMQQueueName,
+      Some(rabbitMQHost).getOrElse("localhost"),
+      Some(rabbitMQPort).getOrElse(DefaultRabbitMQPort),
+      exchangeName,
+      routingKeys,
+      storageLevel)
+  }
+}
+
+private[receiver]
+class RabbitMQReceiver(rabbitMQQueueName: Option[String],
+                       rabbitMQHost: String,
+                       rabbitMQPort: Int,
+                       exchangeName: Option[String],
+                       routingKeys: Seq[String],
+                       storageLevel: StorageLevel)
+  extends Receiver[String](storageLevel) with Logging {
+
+  val DirectExchangeType: String = "direct"
+
+  def onStart() {
+    implicit val akkaSystem = akka.actor.ActorSystem()
+    receive()
+  }
+
+  def onStop() {
+    // There is nothing much to do as the thread calling receive()
+    // is designed to stop by itself isStopped() returns false
+  }
+
+  private def getQueueName(channel: Channel): String ={
+    if (!routingKeys.isEmpty){
+      channel.queueDeclare().getQueue()
+    }else{
+      rabbitMQQueueName.get
+    }
+  }
+
+  /** Create a socket connection and receive data until receiver is stopped */
+  private def receive() {
+
+    try {
```

Comment:
I think this try would be more readable if you change by something like:

for {
  factory <- getConnectionFactory // Try[ConnectionFactory]
  connection <- factory.newConnection // Try[Connection]
  channel <- connection.createChannel
} yield {
  (connection,channel)
}
...


http://danielwestheide.com/blog/2012/12/26/the-neophytes-guide-to-scala-part-6-error-handling-with-try.html